modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinRetrofitCodegen.java

```diff
@@ -0,0 +1,958 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openapitools.codegen.languages;
+
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.Schema;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.*;
+import org.openapitools.codegen.utils.ModelUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.*;
+
+import static org.openapitools.codegen.utils.StringUtils.*;
+
+public class KotlinRetrofitCodegen extends DefaultCodegen implements CodegenConfig {
```

Comment:
I guess `AbstractKotlinCodegen` should be used here

Reply:
Nope. AbstractKotlinCodegen has some wrong implementations. See my explanation [here](https://github.com/OpenAPITools/openapi-generator/blob/76ad145f198467ed160e55950abcf6f6596b2372/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinRetrofitCodegen.java#L189) which is just one error I found in the abstract class.
We could merge the corrections in my Codegen into AbstractKotlinCodegen, though.

Reply:
Fix found error in `AbstractKotlinCodegen` is the right option. Plus, it can be done in separate PR to make contributions smaller -> get reviews easier -> merge faster.

BTW it's a nice contribution, looking forward to it :+1:

Reply:
Agreed with @MatkovIvan to fix the bug in AbstractKotlinCodegen instead.